build/vagrant/infra/userdata.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+# capture output from script
+set -eo
+
+# update machine
+sudo apt-get update
+
+# install needed tools
+sudo apt-get install awscli -y
+sudo apt-get install virtualbox -y
+sudo apt-get install virtualbox-guest-additions-iso -y
+sudo apt-get install vagrant -y
+#sudo apt-get install packer -y
+sudo apt install python3-pip -y
```

Comment:
All of these packages can be installed in one sudo apt-get install line:

`sudo apt-get install awscli virtualbox virtualbox-guest-additions-iso vagrant python3-pip -y`

Or if the debian/ubuntu distro is new enough:

`sudo apt install awscli virtualbox virtualbox-guest-additions-iso vagrant python3-pip -y`